finetuner/tuner/evaluation.py

```diff
@@ -0,0 +1,132 @@
+from typing import TYPE_CHECKING, Optional, List, Dict
+from jina import Document, DocumentArray
+
+from .metrics import METRICS
+from .. import __default_tag_key__
+from ..embedding import embed
+
+if TYPE_CHECKING:
+    from ..helper import AnyDNN, DocumentSequence
+
+
+class Evaluator:
+    def __init__(
+        self,
+        eval_data: 'DocumentSequence',
+        catalog: 'DocumentSequence',
+        embed_model: Optional['AnyDNN'] = None,
+        metrics: Optional[List[str]] = None,
+        distance_metric: str = 'cosine',
+        limit: int = 20,
+    ):
+        """
+        Build an Evaluator object that can be used to evaluate an embedding model on a retrieval task
+
+        :param eval_data: a sequence of documents. Each document should contain ground truth matches under
+            ``doc.matches`` and relevance scores under ``doc.tags[__default_tag_key__]['label']``
+        :param catalog: a sequence of documents, against whist the eval docs will be matched.
+        :param embed_model: the embedding model to use, in order to extract document representations.
+        :param metrics: an optional list of str identifiers specifying the metrics that will be computed. If
+            ``None`` is given, all available metrics will be calculated.
+        :param distance_metric: which distance metric to use when matching documents
+        :param limit: limit the number of results during matching.
+        :return: None.
+        """
+        if metrics:
+            self._metrics = {}
+            for metric in metrics:
+                if metric not in METRICS:
+                    raise ValueError(f"Unknown metric '{metric}'")
+                self._metrics[metric] = METRICS[metric]
```

Comment:
We should either test functionality of just choosing specific metrics or remove the `metrics` argument completely. Thinking about it, I am in favor of removing it and rather add it back in the future.

Reply:
`metrics` argument removed from constructor